src/policy/lockfreeimmortalspace.rs

```diff
@@ -241,15 +242,22 @@ impl<VM: VMBinding> LockFreeImmortalSpace<VM> {
             *args.options.transparent_hugepages,
             crate::util::memory::MmapProtection::ReadWrite,
         );
-        crate::util::memory::dzmmap_noreplace(start, aligned_total_bytes, strategy).unwrap();
-        if space
+        crate::util::memory::dzmmap_noreplace(
+            start,
+            aligned_total_bytes,
+            strategy,
+            &MmapAnnotation::Space {
+                name: space.get_name(),
+            },
+        )
+        .unwrap();
+        space
             .metadata
-            .try_map_metadata_space(start, aligned_total_bytes)
-            .is_err()
-        {
-            // TODO(Javad): handle meta space allocation failure
-            panic!("failed to mmap meta memory");
-        }
+            .try_map_metadata_space(start, aligned_total_bytes, space.get_name())
+            .unwrap_or_else(|e| {
```

Comment:
I don't see improvement from this refactoring. This refactoring seems based on a personal coding style, rather than improvement for the code, which is not a reason to refactor existing code base. We have style checks and lint checks in CI, and any code that passed the checks is considered acceptable. We should avoid refactoring existing code purely based on personal coding style. Alternatively, a better approach is to introduce new lints (and justify why we need the new lints) so we adopt new styles as our 'standard' styles.

Besides, here we just need to know if the call returns an error, and we do not really need the value of the result. Using `is_err` is okay, or even better than `unwrap` to reflect what we are doing. 

I don't request the refactoring to be reverted (it doesn't worth the extra efforts), but refactoring purely based on personal coding style should be discouraged.

Reply:
The intention is showing the error in the panic message.  There are several things that can go wrong in `mmap_fixed`, namely `mmap`, `madvise` and the newly introduced `prctl`.  Here we don't really need the value of the result for the correct functioning of the program (because we just panic).  But if we omit the error, it will simply print "failed to mmap meta memory" in the panic message, and the stack trace will not include the exact place that returned the error.  That will leave the user or developer wondering what went wrong.  Showing the error message at least will provide the user some hints about the error.

I considered letting `prctl` return `Err`, too, because it does fail in some cases, for example, the name string is too long.  But since it cannot be distinguished from the case when it is executed on a kernel older than 5.17, I chose to log the error rather than returning `Err` or panicking.